### PR TITLE
Parse ASCII status chars and add SAE states

### PR DESCRIPTION
**Pull request by Moustachauve**
Enhance Status handling to accept the TechnoVE API's single-character ASCII status values (or their integer ordinals) and normalise them to enum members. Add missing SAE J1772 states (D/E/F) and explicit TechnoVE states, update mappings to use ord(...) constants, and improve the Status.build() docstring and type handling. Expand tests to cover string inputs, new states, and edge cases (unknown ints/strings/None).

### CODE REVIEW

File: src/technove/models.py [hunk 1]
```diff
@@ -7,33 +8,78 @@
...
+            ord("H"): Status.GROUND_FAULT,  # 72
+            ord("S"): Status.OUT_OF_ACTIVATION_PERIOD,  # 83
+            ord("T"): Status.HIGH_TARIFF_PERIOD,  # 84
         }
 
-        if status in statuses:
-            return statuses[status]
-
-        return Status.UNKNOWN
+        return statuses.get(code, Status.UNKNOWN)
```
[the remark this answers is not in the document]
Comment (Moustachauve) [reply src/technove/models.py]: Done.
